src/Language/PureScript/TypeChecker/Entailment.hs

```diff
@@ -334,6 +345,28 @@ entails SolverOptions{..} constraint context hints =
         subclassDictionaryValue dict className index =
           App (Accessor (mkString (superclassName className index)) dict) valUndefined
 
+    -- | Left biased union of two row types
+    unionRows :: Type -> Type -> Type -> Maybe (Type, Type, Type, Maybe [Constraint])
+    unionRows l r _ =
+        guard canMakeProgress $> (l, r, rowFromList out, cons)
+      where
+        (fixed, rest) = rowToList l
+
+        rowVar = TypeVar "r"
```

Comment:
I believe this is safe. This is like the user introducing a `TypeVar` in an instance declaration. By the time we get to the solver, everything else will be skolemized or instantiated.